lib/language.js

```diff
@@ -7,42 +7,40 @@
 const locale = require('locale')
 
 const cookieName = 'language'
+const cookieOptions = { secure: true, sameSite: 'strict', httpOnly: true }
 const validLanguages = ['sv', 'en']
 const defaultLanguage = validLanguages[0]
 const supportedLocales = new locale.Locales(validLanguages, defaultLanguage)
 
+const validLanguage = lang => {
+  if (validLanguages.includes(lang)) {
+    return lang
+  }
+}
+
+const useAcceptLanguageOrDefault = preferredLanguages => {
+  const preferredLocales = new locale.Locales(preferredLanguages, defaultLanguage)
+  const bestLocale = preferredLocales.best(supportedLocales)
+  return bestLocale.language
+}
+
 /**
  * Initialize locale and language for the current user session (respects cookie: language).
  */
 function _init(req, res, newLang) {
-  let lang = newLang || req.cookies[cookieName]
+  const lang =
+    validLanguage(newLang) ||
+    validLanguage(req.cookies[cookieName]) ||
+    useAcceptLanguageOrDefault(req.headers['accept-language'])
 
-  // Only allow lang to be one of the valid languages
-  if (lang && !validLanguages.includes(lang)) lang = defaultLanguage
-
-  let chosenLocale
-
-  if (lang) {
-    const locales = new locale.Locales(lang, defaultLanguage)
-    chosenLocale = locales.best(supportedLocales)
-  } else {
-    const locales = new locale.Locales(req.headers['accept-language'], defaultLanguage)
-    chosenLocale = locales.best(supportedLocales)
+  if (req.cookies[cookieName] !== lang) {
+    res.cookie(cookieName, lang, cookieOptions)
   }
 
-  // If we got an explicit language we set the language cookie
-  if (newLang) {
-    res.cookie(cookieName, chosenLocale.language, { secure: true, sameSite: 'strict', httpOnly: true })
-  } else if (!req.cookies[cookieName]) {
-    // Make sure language cookie is set so subsequent requests are guaranteed to use the same language
-    res.cookie(cookieName, lang || 'sv', { secure: true, sameSite: 'strict', httpOnly: true })
-  }
-
-  res.locals.locale = chosenLocale
+  const locales = new locale.Locales(lang)
+  res.locals.locale = locales
   // Backwards compatibility only in case someone accessed the prop directly:
-  res.locals.language = chosenLocale.language
-
-  return chosenLocale
+  res.locals.language = locales.language
```

Comment:
Just a thought: Should this be `res.locales` or is it referring to something `local` in `res`?

Reply:
It is referring to this: https://expressjs.com/en/api.html#res.locals. 👍

Reply:
TIL